src/errors/NetworkError.ts

```diff
@@ -0,0 +1,22 @@
+import type { ErrorLike } from "@apollo/client";
+import { CombinedGraphQLErrors } from "@apollo/client";
+
+const registry = new WeakSet();
+
+/** @internal Please do not use directly. */
+export function registerNetworkError(error: ErrorLike) {
+  if (!CombinedGraphQLErrors.is(error)) {
+    registry.add(error);
```

Comment:
```suggestion
const networkErrorSymbol = Symbol.for("apollo.networkError");

/** @internal Please do not use directly. */
export function registerNetworkError(error: ErrorLike) {
  if (!CombinedGraphQLErrors.is(error)) {
    if (Object.isExtensible(error)) {
      Object.defineProperty(error, networkErrorSymbol, {
        value: true,
        enumerable: false,
        writable: false,
        configurable: true,
      });
    } else {
      registry.add(error);
    }
```

Something like this could be a memory-defensive version that only uses `registry` as a last resort and tries to "mark" errors first.

On the other hand, I might be way overhinking this, too. Environments where `WeakSet` needs to be polyfilled are probably not that common, and where I can imagine they exist (e.g. edge runtimes), they probably also are not very long-lived.

We really need a list of alternative JS runtimes and their features at some point.

Reply:
Let's keep this in our back pocket. If we end up in situations where the `WeakSet` becomes a problem, using that defensive fallback is a great solution.